### DISCUSSION TITLE
How to handle DB2 and return an AST syntax tree

### DISCUSSION THREAD

**Discussion by HyDreamFly** [antlr/grammars-v4#4091] [Q&A]
Is there a grammar file available

**Comment by kaby76** [comment]
There's no DB2 grammar here. But, there are some elsewhere.

* https://www.reddit.com/r/mainframe/comments/psjwco/antlr_grammars_for_jcl_cobol_embedded_db2_sql/
* https://medium.com/@craig.schneiderwent/mainframe-application-portfolio-analysis-3ee859c62225
* https://github.com/cschneid-the-elder/mapa/tree/695663f0e445bc3c7c77c788ac9482bea14278e7/db2z/src

Antlr4 does not construct ASTs. It constructs a CST, which is arguably better in so many ways.

**Reply by HyDreamFly** [reply]
thank you！May I ask another question，Do you understand GaussDB，Is there a grammar file available？
